test262/per-file/main.js

```diff
@@ -0,0 +1,225 @@
+"use strict";
+
+// (() => {
+let resultsNode;
+let summaryLabel;
+let summaryStatusLabel;
+let leafTreeNodeTemplate;
+let nonLeafTreeNodeTemplate;
+let pathInTree = [];
+let tree;
+const resultsObject = Object.create(null);
+
+function initialize(data, modeName) {
+  let mode;
+  if (modeName === "") {
+    mode = "AST";
+  } else if (modeName === "-bytecode") {
+    mode = "bytecode";
+  } else {
+    throw new Error("Unknown mode: " + modeName);
+  }
+
+  // Do a pass and generate the tree.
+  for (const testPath in data.results) {
+    const segments = testPath.split("/");
+    const fileName = segments.pop();
+    let testObject = resultsObject;
+    for (const pathSegment of segments) {
+      if (!(pathSegment in testObject)) {
+        testObject[pathSegment] = {
+          children: Object.create(null),
+          aggregatedResults: null,
+        };
+      }
+
+      testObject = testObject[pathSegment].children;
+    }
+
+    if (!(fileName in testObject)) {
+      testObject[fileName] = {
+        children: null,
+        results: Object.create(null),
+      };
+    }
+
+    testObject[fileName].results[mode] = data.results[testPath].toLowerCase();
+  }
+}
+
+function generateResults() {
+  function constructTree(results) {
+    if (results.children === null) {
+      results.aggregatedResults = Object.fromEntries(
+        Object.keys(results.results).map((name) => [
+          name,
+          { [results.results[name]]: 1 },
+        ])
+      );
+
+      return;
+    }
+
+    for (const name in results.children) constructTree(results.children[name]);
```

Comment:
I hate prettier for doing this, just why
It still doesn't have an option to break without curlies, so if you think this looks crap add them (even if it's just a single line body) - up to you.

Reply:
I do, and I really wanted to add curlies
I'll do that : )